lib/takeaway.rb

```diff
@@ -0,0 +1,22 @@
+require_relative 'menu'
+
+class Takeaway
+  attr_reader :order
+
+  @@menu = Menu.new
+
+  def initialize
+  @order = []
+  end
+
+  def add_dish(dish)
+    fail "Dish not on menu!" if !@@menu.dishes.key?(dish)
+    @order << dish
+  end
+
+  def total
```

Comment:
This is nice and neat.  Since it's possible to pass a block to sum, there might be a more concise way of writing this though.

How about...

```ruby
@order.sum { |dish| @@menu.dishes[dish] }
```

Reply:
Which do you prefer in terms of readability?